### PR TITLE
Add more task debug info in jobInfo api response [rest-server]

### PR DESCRIPTION
**Pull request by debuggy**

### CODE REVIEW

**Comment by yqwang-ms** [comment]
>       // for backward compatibility

Add doc for the new fields?
I have gave their explanation in onenote, you can refine it in PAI doc?

---
Refers to: src/rest-server/src/models/v2/job/k8s.js:204 in 71c04eb. [](commit_id = 71c04ebe966d75ca0bd9724d57f10aa9e15f630e, deletion_comment = False)

**Comment by yqwang-ms** [comment]
>       virtualCluster,

Seems appExitType is useless and duplicated with appExitSpec, should we remove it?

BTW, the completionStatus.type.name only have 2 values;
```
type CompletionType struct {
 Name       CompletionTypeName        `json:"name" yaml:"name"`
 Attributes []CompletionTypeAttribute `json:"attributes" yaml:"attributes"`
}

type CompletionTypeName string

const (
 CompletionTypeNameSucceeded CompletionTypeName = "Succeeded"
 CompletionTypeNameFailed    CompletionTypeName = "Failed"
)


```


---
Refers to: src/rest-server/src/models/v2/job/k8s.js:303 in 71c04eb. [](commit_id = 71c04ebe966d75ca0bd9724d57f10aa9e15f630e, deletion_comment = False)

File: src/rest-server/src/models/v2/job/k8s.js [hunk 1]
```diff
@@ -222,10 +224,16 @@ const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
     ),
     containerId: taskStatus.attemptStatus.podUID,
     containerIp: taskStatus.attemptStatus.podHostIP,
+    podNodeName: taskStatus.attemptStatus.podNodeName,
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job/k8s.js:227]: >podNodeName [](start = 4, length = 11)

-> containerNodeName?

File: src/rest-server/src/models/v2/job/k8s.js [hunk 2]
```diff
@@ -222,10 +224,16 @@ const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
...
     containerPorts,
     containerGpus,
     containerLog: `http://${taskStatus.attemptStatus.podHostIP}:${process.env.LOG_MANAGER_PORT}/log-manager/tail/${logPathPrefix}/${taskStatus.attemptStatus.podUID}/`,
     containerExitCode: completionStatus ? completionStatus.code : null,
+    containerExitSpec: completionStatus ? generateExitSpec(completionStatus.code) : generateExitSpec(null),
+    containerExitDiagnostics: exitDiagnostics ? exitDiagnostics.diagnosticsSummary : null,
+    totalRetriedCount: taskStatus.retryPolicyStatus.totalRetriedCount,
+    startTime: taskStatus.attemptStatus.startTime,
+    completionTime: taskStatus.attemptStatus.completionTime,
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job/k8s.js:236]: Align to job time fields, such as;

```
      createdTime: new Date(taskStatus.StartTime).getTime(),
      completedTime: new Date(taskStatus.completionTime).getTime(),
      attemptLaunchedTime: new Date(taskStatus.attemptStatus.runTime || taskStatus.attemptStatus.completionTime).getTime(),
      attemptCompletedTime: new Date(taskStatus.attemptStatus.completionTime).getTime(),
````

File: src/rest-server/src/models/v2/job/k8s.js [hunk 3]
```diff
@@ -222,10 +224,16 @@ const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
...
     containerIp: taskStatus.attemptStatus.podHostIP,
+    podNodeName: taskStatus.attemptStatus.podNodeName,
     containerPorts,
     containerGpus,
     containerLog: `http://${taskStatus.attemptStatus.podHostIP}:${process.env.LOG_MANAGER_PORT}/log-manager/tail/${logPathPrefix}/${taskStatus.attemptStatus.podUID}/`,
     containerExitCode: completionStatus ? completionStatus.code : null,
+    containerExitSpec: completionStatus ? generateExitSpec(completionStatus.code) : generateExitSpec(null),
+    containerExitDiagnostics: exitDiagnostics ? exitDiagnostics.diagnosticsSummary : null,
+    totalRetriedCount: taskStatus.retryPolicyStatus.totalRetriedCount,
+    startTime: taskStatus.attemptStatus.startTime,
```
Comment (yqwang-ms) [inline src/rest-server/src/models/v2/job/k8s.js:235]: Align to job retry fields:

```
      retries: taskStatus.retryPolicyStatus.totalRetriedCount,
      retryDetails: {
        user: taskStatus.retryPolicyStatus.accountableRetriedCount,
        platform: taskStatus.retryPolicyStatus.totalRetriedCount - taskStatus.retryPolicyStatus.accountableRetriedCount,
        resource: 0,
      },
```

**Comment by debuggy** [comment]
> 
> 
> > ```
> >   virtualCluster,
> > ```
> 
> Seems appExitType is useless and duplicated with appExitSpec, should we remove it?
> 
> BTW, the completionStatus.type.name only have 2 values;
> 
> ```
> type CompletionType struct {
>  Name       CompletionTypeName        `json:"name" yaml:"name"`
>  Attributes []CompletionTypeAttribute `json:"attributes" yaml:"attributes"`
> }
> 
> type CompletionTypeName string
> 
> const (
>  CompletionTypeNameSucceeded CompletionTypeName = "Succeeded"
>  CompletionTypeNameFailed    CompletionTypeName = "Failed"
> )
> ```
> 
> Refers to: src/rest-server/src/models/v2/job/k8s.js:303 in [71c04eb](https://github.com/microsoft/pai/commit/71c04ebe966d75ca0bd9724d57f10aa9e15f630e). [](commit_id = [71c04eb](https://github.com/microsoft/pai/commit/71c04ebe966d75ca0bd9724d57f10aa9e15f630e), deletion_comment = False)



> 
> 
> > ```
> >   virtualCluster,
> > ```
> 
> Seems appExitType is useless and duplicated with appExitSpec, should we remove it?
> 
> BTW, the completionStatus.type.name only have 2 values;
> 
> ```
> type CompletionType struct {
>  Name       CompletionTypeName        `json:"name" yaml:"name"`
>  Attributes []CompletionTypeAttribute `json:"attributes" yaml:"attributes"`
> }
> 
> type CompletionTypeName string
> 
> const (
>  CompletionTypeNameSucceeded CompletionTypeName = "Succeeded"
>  CompletionTypeNameFailed    CompletionTypeName = "Failed"
> )
> ```
> 
> Refers to: src/rest-server/src/models/v2/job/k8s.js:303 in [71c04eb](https://github.com/microsoft/pai/commit/71c04ebe966d75ca0bd9724d57f10aa9e15f630e). [](commit_id = [71c04eb](https://github.com/microsoft/pai/commit/71c04ebe966d75ca0bd9724d57f10aa9e15f630e), deletion_comment = False)

If we delete a previous property, I am not sure any side effects on frontend. I have checked that webportal does not use this field, but still need to make sure sdk/vscode parts.

File: src/rest-server/docs/swagger.yaml [hunk 4]
```diff
@@ -2079,6 +2088,68 @@ components:
...
+                        Epoch.
+                    completedTime:
+                      type: integer
+                      description: >-
+                        task completion time, in number of milliseconds since the Unix
+                        Epoch.
+                    currentAttemptLaunchedTime:
+                      type: integer
+                      description: >-
+                        the last attempt launched time, in number of milliseconds since the Unix
```
Comment (yqwang-ms) [inline src/rest-server/docs/swagger.yaml:2143]: >the last attempt [](start = 24, length = 16)

current attempt launched time

File: src/rest-server/docs/swagger.yaml [hunk 5]
```diff
@@ -2065,6 +2071,9 @@ components:
                       type: string
                       nullable: true
                       description: ip of the task container
+                    containerNodeName:
+                      type: string
```
Comment (yiyione) [inline src/rest-server/docs/swagger.yaml:2075]: May return null, if the job still waiting.

**Pull request merged**